src/OneExplorer/OneExplorer.ts

```diff
@@ -163,6 +163,50 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
     this._onDidChangeTreeData.fire();
   }
 
+  // TODO: Add move()
+
+  /**
+   * Rename a file of all types of nodes (baseModel, derivedModel, config) excepts for directory.
+   * It only alters the file name, not the path.
+   */
+  rename(oneNode: OneNode): void {
+    // TODO: prohibit special characters for security ('..', '*', etc)
+    let warningMessage;
+    if (oneNode.node.type === NodeType.baseModel) {
+      // TODO automatically change the corresponding files
+      warningMessage = 'WARNING: You may need to change input paths in the following cfg files:\n';
+      for (const conf of oneNode.node.childNodes) {
+        warningMessage += `${conf.name} \n`;
```

Comment:
(Q) Does quick input title support new line character?

Reply:
Nope.

Reply:
https://stackoverflow.com/a/57735348/7949783

They say vscode team are explicitly removing \n and they will not do support them. I may remove this.

Reply:
https://github.com/microsoft/vscode/issues/48900#issuecomment-385332787